src/lib/codap-helper.ts

```diff
@@ -315,16 +331,18 @@ export class CodapHelper {
       }
 
       // After initial join we allow destructive syncing
-      if (!initialJoin) {
-        const staleAttributes = originalAttributes.filter(attrA => {
-          return !sharedAttributes.some(attrB => attrA.name === attrB.name);
-        });
-
-        changeCommands.push(...staleAttributes.map(attr => ({
-          action: "delete",
-          resource: collectionResource(dataContext.name, attr.collection, `attribute[${attr.name}]`)
-        })));
-      }
+      // Disabled for now, as we still have echo effects that sometimes
+      // cause attributes to be deleted incorrectly on initial join.
```

Comment:
Encountered more echo/rebound effects resulting in deleting of merged attributes. Disabling deletion of attributes on sync for the moment as correct merging is more important than deleting attributes on sync right now.